plugins/uefi-dbx/fu-uefi-dbx-device.c

```diff
@@ -165,6 +166,14 @@ fu_uefi_dbx_device_probe(FuDevice *device, GError **error)
 						 NULL);
 		fu_device_build_instance_id(device, NULL, "UEFI", "CRT", "ARCH", NULL);
 	}
+
+	/* dbx changes are expected to change PCR7, warn the user that BitLocker might ask for
+	recovery key after fw update */
+	if (!fu_common_check_full_disk_encryption(&error_fde)) {
+		g_debug("FDE in use, set flag: %s", error_fde->message);
+		fu_device_add_flag(device, FWUPD_DEVICE_FLAG_AFFECTS_FDE);
+	}
```

Comment:
for how the codebase works "today" this makes sense, but I have to question if we should be changing semantics of `fu_common_check_full_disk_encryption` as part of this PR.

Right now it returns `TRUE` for no FDE and `FALSE` for a failure to get disks **or** FDE found.

I think there should be a gboolean pointer sent as an argument that has the result of the check.  Something like:

```
gboolean
fu_common_check_full_disk_encryption(gboolean *result, GError **error)
```

Then return `TRUE` if the function "was checked" successfully, `FALSE` if there was a disk problem.  And `*result` can be set based upon whether FDE was found or not.

Reply:
yup, lets merge this and do that. @kyl191 are you up for a follow on PR? :)

Reply:
@hughsie yep! 

Bit rusty with C but I follow @superm1's idea - changes don't sound too invasive and worst case I can try running it in valgrind.